balancer/catabalancer/catalyst_balancer.go

```diff
@@ -88,6 +119,7 @@ func NewBalancer(nodeName string, metricTimeout time.Duration, ingestStreamTimeo
 		NodeMetrics:         make(map[string]NodeMetrics),
 		metricTimeout:       metricTimeout,
 		ingestStreamTimeout: ingestStreamTimeout,
+		MetricsDB:           nodeStatsDB,
```

Comment:
IIUC This will be a different database than our metrics DB. Then, could you rename it to something like `NodeStatsDB`?